src/parser_json.c

```diff
@@ -290,7 +290,7 @@ lydjson_data_skip(struct lyjson_ctx *jsonctx)
         LY_CHECK_RET(lyjson_ctx_next(jsonctx, &current));
         if (current == status) {
             sublevels++;
-        } else if (current == status + 1) {
+        } else if (current == status + 1 || current == status + 2) {
```

Comment:
Every boolean condition should be in parentheses so for tests to pass this should be
```C
} else if ((current == status + 1) || (current == status + 2)) {
```